File: MCP服务器/developer-tools/TeamDman_mcp-server-hello_0a3d3df_rust/src/mcp/tools.rs
use crate::mcp::types::*;
use maplit::hashmap;
use rpc_router::Handler;
use rpc_router::HandlerResult;
use rpc_router::RouterBuilder;
use rpc_router::RpcParams;
use serde::Deserialize;
use serde::Serialize;

/// register all tools to the router
pub fn register_tools(router_builder: RouterBuilder) -> RouterBuilder {
    router_builder
        .append_dyn("tools/list", tools_list.into_dyn())
        .append_dyn("get_current_time_in_city", current_time.into_dyn())
}

pub async fn tools_list(_request: Option<ListToolsRequest>) -> HandlerResult<ListToolsResult> {
    //let tools: Vec<Tool> = serde_json::from_str(include_str!("./templates/tools.json")).unwrap();
    let response = ListToolsResult {
        tools: vec![Tool {
            name: "get_current_time_in_city".to_string(),
            description: Some("Get the current time in the city".to_string()),
            input_schema: ToolInputSchema {
                type_name: "object".to_string(),
                properties: hashmap! {
                    "city".to_string() => ToolInputSchemaProperty {
                        type_name: Some("string".to_owned()),
                        description: Some("city name".to_owned()),
                        enum_values: None,
                    }
                },
                required: vec!["city".to_string()],
            },
        }],
        next_cursor: None,
    };
    Ok(response)
}

#[derive(Deserialize, Serialize, RpcParams)]
pub struct CurrentTimeRequest {
    pub city: Option<String>,
}

pub async fn current_time(_request: CurrentTimeRequest) -> HandlerResult<CallToolResult> {
    let result = format!("Now: {}!", chrono::Local::now().to_rfc2822());
    Ok(CallToolResult {
        content: vec![CallToolResultContent::Text { text: result }],
        is_error: false,
    })
}

File: reference-mcp-servers/Cloud_Platforms/qiniu_qiniu-mcp-server_52dd335_python/src/mcp_server/core/version/tools.py

from mcp import types

from . import version
from ...tools import tools


class _ToolImpl:
    def __init__(self):
        pass

    @tools.tool_meta(
        types.Tool(
            name="version",
            description="qiniu mcp server version info.",
            inputSchema={
                "type": "object",
                "required": [],
            }
        )
    )
    def version(self, **kwargs) -> list[types.TextContent]:
        return [types.TextContent(type="text", text=version.VERSION)]

def register_tools():
    tool_impl = _ToolImpl()
    tools.auto_register_tools(
        [
            tool_impl.version,
        ]
    )